src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -6595,6 +6605,27 @@ class StubGenerator: public StubCodeGenerator {
   }
 #endif // LINUX
 
+  // Pass object argument in r0 (which has to be preserved outside this stub)
+  // Pass back result in r0
+  // Clobbers rscratch1
+  address generate_load_nklass() {
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "load_nklass");
+
+    address start = __ pc();
+
+    __ set_last_Java_frame(sp, rfp, lr, rscratch1);
+    __ enter();
+    __ push_call_clobbered_registers_except(r0);
+    __ call_VM_leaf(CAST_FROM_FN_PTR(address, oopDesc::load_nklass_runtime), 1);
```

Comment:
I'm tempted to suggest saving flags here too. But I guess you've checked all the places that use `load_klass()`. At least the comment for `load_klass()` should advertise exactly what gets clobbered.

Reply:
Hmm yeah, maybe. I found one instance where flags are live around this call, but could work around it:

https://github.com/rkennke/lilliput/blob/86c13e4dfc6c1e53c9c97fc686d299c4615fd28d/src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp#L587

Also, it's not only the stub that clobbers the condition flags, but the fast-path, too.

Reply:
Fast path is fixable.